### PR TITLE
Add content-type header to all outbound calls from MLFlow client

### PR DESCRIPTION
**Pull request by WillEngler**
Resolves [#11](https://github.com/Garden-AI/garden-backend/issues/11). Sibling PR with https://github.com/Garden-AI/garden-backend/pull/12

## Overview

Adds Content-Type HTTP header with value "application/octet-stream" to all outbound requests generated by the MLFlow client. This enables us to specify in AWS API Gateway that we do not want to coerce payloads with that header to text.

## Discussion

This adds code that accesses a protected attribute of the MLFlow library, which is bad. For what it's worth, the `RequestHeaderProvider` registry is intended as an extension point. MLFlow would just prefer you create a new package to do so. I'd like to avoid making a new package or packaging a short-lived fork of MLFlow to unblock us. I think this quick-and-somewhat-dirty fix is a good expedient while we pursue a proper patch into MLFlow.

## Testing

I used the end-to-end test case that Owen wrote (and is included in this PR) to rapidly test this. It confirms that we can send a model up and pull it down without corrupting it.

----
:books: Documentation preview :books:: https://garden-ai--80.org.readthedocs.build/en/80/

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -45,6 +45,5 @@ jobs:
           python -m coverage run -m pytest -m "not integration" -r sx
           codecov --token=${{ secrets.CODECOV_TOKEN }}
       - name: Check for vulnerable libraries with safety
-        # Context on skipped vulnerability report. (It is very minor) https://github.com/sqlalchemy/sqlalchemy/pull/8563#issuecomment-1381987276
         run: |
-          safety check -i 51668
+          safety check
```
Comment (WillEngler) [inline .github/workflows/ci.yaml:49]: Newer version of MLFlow client uses newer version of SQLAlchemy, so we're not triggering warning 51668 any more.

File: garden_ai/mlflow_bandaid/binary_header_provider.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    RequestHeaderProvider,
+)
+
+
+class BinaryContentTypeHeaderProvider(RequestHeaderProvider):
+    """
+    Adds a Content-Type header so that we can ask AWS API Gateway not to mangle our requests.
+    """
+
+    def in_context(self):
```
[the remark this answers is not in the document]
Comment (OwenPriceSkelly) [reply garden_ai/mlflow_bandaid/binary_header_provider.py:11]: I think we should add something to a docstring that gives a little context for the hack like a link to a relevant issue upstream (maybe even just a summary of or link to your slack thread with ben g), partly for like our own sake but more for contributors who might be confused/alarmed by a `mlflow_bandaid` module

[the remark this answers is not in the document]
Comment (OwenPriceSkelly) [reply garden_ai/mlflow_bandaid/binary_header_provider.py:11]: something like the `garden_json_encoder` docstring in utils

**Review by OwenPriceSkelly — CHANGES_REQUESTED** [review]

File: garden_ai/mlflow_bandaid/binary_header_provider.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (WillEngler) [reply garden_ai/mlflow_bandaid/binary_header_provider.py:14]: Good suggestions. I added more context and issue links.

**Review by OwenPriceSkelly — APPROVED** [review]

**Pull request merged**